packages/medusa/src/services/token.ts

```diff
@@ -0,0 +1,47 @@
+import jwt, { Jwt, JwtPayload, SignOptions, VerifyOptions } from "jsonwebtoken"
+import { ConfigModule } from "../types/global"
+import formatRegistrationName from "../utils/format-registration-name"
+import { resolve } from "path"
+import { MedusaError } from "medusa-core-utils"
+
+type InjectedDependencies = {
+  configModule: ConfigModule
+}
+
+class TokenService {
+  static RESOLUTION_KEY = formatRegistrationName(resolve(__dirname, __filename))
+
+  protected configModule_: ConfigModule
```

Comment:
nit(non-blocking): Read only
```suggestion
  protected readonly configModule_: ConfigModule
```